File: _CodeTopics/LeetCode_contest/weekly/weekly2020/206/206_1.py
class Solution(object):
    def numSpecial(self, mat):
        """
        :type mat: List[List[int]]
        :rtype: int
        """
        
        rows, cols = len(mat), len(mat[0])
        res = 0
        
        def sum_of_column(j):
            ret = 0
            for i in range(rows):
                ret += mat[i][j]
            return ret
        
        for i in range(rows):
            if sum(mat[i]) != 1:
                continue
            for j in range(cols):
                if mat[i][j] != 1:
                    continue
                if sum_of_column(j) == 1:
                    res += 1
        return res
    
"""
https://leetcode-cn.com/submissions/detail/107496893/

95 / 95 个通过测试用例
状态：通过
执行用时: 32 ms
内存消耗: 12.8 MB
"""

File: _CodeTopics/LeetCode/1401-1600/001582--weekly206_1/tran/206_1.cpp
class Solution {
public:
    int sum_of_column(vector<vector<int>>& mat, int j) {
        int ret = 0;
        for (int i = 0; i < mat.size(); ++i) {
            ret += mat[i][j];
        }
        return ret;
    }

    int numSpecial(vector<vector<int>>& mat) {

        // 第 206 场周赛第一题
        // https://leetcode-cn.com/submissions/detail/107496893/

        int m = mat.size(), n = mat[0].size();
        int res = 0;

        for (int i = 0; i < m; ++i) {
            if (std::accumulate(mat[i].begin(), mat[i].end(), 0) != 1) {
                continue;
            }
            for (int j = 0; j < n; ++j) {
                if (mat[i][j] != 1) {
                    continue;
                }
                if (sum_of_column(mat, j) == 1) {
                    res += 1;
                }
            }
        }
        return res;
    }
};

/*
https://leetcode.cn/submissions/detail/359082129/

执行用时：
20 ms
, 在所有 C++ 提交中击败了
49.00%
的用户
内存消耗：
12.5 MB
, 在所有 C++ 提交中击败了
60.64%
的用户
通过测试用例：
95 / 95
*/
